crates/astria-sequencer/src/service/consensus.rs

```diff
@@ -80,44 +83,40 @@ impl Consensus {
                     .context("failed initializing chain")?,
             ),
             ConsensusRequest::PrepareProposal(prepare_proposal) => {
-                ConsensusResponse::PrepareProposal(response::PrepareProposal {
-                    txs: prepare_proposal.txs,
-                })
-            }
-            ConsensusRequest::ProcessProposal(_process_proposal) => {
-                // TODO: handle this
-                ConsensusResponse::ProcessProposal(response::ProcessProposal::Accept)
+                ConsensusResponse::PrepareProposal(
+                    handle_prepare_proposal(prepare_proposal)
+                        .context("failed to prepare proposal")?,
```

Comment:
What happens if this returns with an error? I tried tracing the error back:

1. The loop inside `Consensus::run` creates a `warn!` event, and then sends the `Result<ConsensusResponse, BoxError>` back to the tower-abci server.
2. `tower-abci::Server` pushes the consensus `call` futures into a `FuturesUnordered`: https://docs.rs/tower-abci/latest/src/tower_abci/v037/server.rs.html#207
3. `tower-abci::Server` polls `FuturesUnordered` until it sees the consensus response https://docs.rs/tower-abci/latest/src/tower_abci/v037/server.rs.html#244
    a. the `response?.into()` lines tells me that it short circuits and just takes down the entire server if consensus encounters an issue?

This seems harsh. Reading `CometBFT` docs on [`PrepareProposal`](https://docs.cometbft.com/v0.37/spec/abci/abci++_methods#parameters-and-types) (scroll down a bit) I found this:
> If the Application considers that tx should not be proposed in this block, e.g., there are other transactions with higher priority, then it should not include it in ResponsePrepareProposal.txs. However, this will not remove tx from the mempool.

So wouldn't it be ok to just returns a `ConsensusResponse` with `txs` left empty?

Reply:
originally I thought this would be ok, as txs in the tendermint mempool go through `check_tx` first which will return an error code if the bytes fail to deserialized into `Signed`, so nothing passed from the mempool to the app should be un-deserializable. however I agree this does seem too dramatic in case somehow an error does occur; I updated it to ignore any tx bytes that aren't deserializable into `Signed`, so there is now no error returned.